pyccel/codegen/printing/ccode.py

```diff
@@ -1186,25 +1203,34 @@ def _print_OMP_Single_Construct(self, expr):
     def _print_Omp_End_Clause(self, expr):
         return '}'
     #=====================================
+    def free_allocs(self, variables):
+
+        free_str = ''
+        for i in variables:
+            free_str += '\nfree_array(%s);' % i.name
+        return free_str
```

Comment:
This can be written more simply as:
```
free_str = '\n'.join('free_array({});'.format(i.name) for i in variables)
```
This has 2 benefits:
1. The expression no longer starts with an unnecessary line break
2. Generator expressions are faster than for loops